engine/base.ftl

```diff
@@ -831,6 +834,56 @@ are added.
                 [/#if]
             [/#list]
 
+            [#-- Process "AllowAdditionalParameters" value   --]
+            [#-- false: the matched candidated object cannot --]
+            [#--        have undefined parameters provided.  --]
+            [#if attribute.Children?has_content && providedCandidate?has_content && providedValue?is_hash]
+                [#if !(attribute.AllowAdditionalParameters)]
+                    [#local validKeys = asFlattenedArray(attribute.Children?map(c -> c.Names))]
+                    [#if attribute.SubObjects || attribute.Subobjects!false]
+                        [#local validationObjects = providedValue?values]
+                    [#else]
+                        [#local validationObjects = [providedValue]]
+                    [/#if]
+                    [#list validationObjects as validationObject]
+                        [#local parameters = validationObject?keys]
+                        [#-- Common Parameters that are used throughout        --]
+                        [#-- but are as-yet unaccounted for in the             --]
+                        [#-- composite object definitions.                     --]
+                        [#-- TODO(rossmurr4y): add commonParams to definitions --]
+                        [#local commonParams = [
+                            "Id",
+                            "Name",
+                            "Title",
+                            "Enabled",
+                            "Description",
+                            "Instances",
+                            "Configured",
+                            "Extensions",
+                            "multiAZ",
+                            "deployment:Unit",
+                            "deployent:Group"]
```

Comment:
These should be added to the appropriate composite objects, weird that they aren't 
It also includes `deployment:Priority` 

Should this also include `DeploymentUnits`